File: src/Report/Report.java
package Report;

import java.io.*;
import java.nio.charset.StandardCharsets;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class Report {
	private DateTimeFormatter date = DateTimeFormatter.ofPattern("EEEE MMMM dd, yyyy HH:mm:ss");
	private LocalDateTime now = LocalDateTime.now();
	private FileWriter writeTable = new FileWriter("src/resources/Report.txt", StandardCharsets.UTF_8);
	private BufferedWriter bw = new BufferedWriter(writeTable);
	private PrintWriter out = new PrintWriter(bw);
	private BufferedReader supplier = new BufferedReader(new FileReader("Supplier Info.txt"));
	private BufferedReader customer = new BufferedReader(new FileReader("Customer Info.txt"));
	private BufferedReader product = new BufferedReader(new FileReader("Product.txt"));


	/*Report(ExternalEggSupplier externalEggSupplier, Customer customerArray,Product product, Egg egg) throws IOException {
		this.externalEggSupplier = externalEggSupplier;
		this.customerArray = customerArray;
		this.product = product;
		this.egg = egg;
	}*/
	public Report() throws IOException {

	}

	public void generateReport() throws IOException {
		out.println("Created by " + "Supervisor ");
		out.println("Compiled on: " + getDate().format(now) + "\n\n");
		String count;
		while ((count = supplier.readLine()) != null) {
			out.println(count);
		}
		out.println("\n");
		while ((count = customer.readLine()) != null) {
			out.println(count);
		}
		out.println("\n");

		while ((count = product.readLine()) != null) {
			out.println(count);
		}
		out.flush();
		out.close();
		supplier.close();
	}


	private DateTimeFormatter getDate() {
		return date;
	}


}

File: src/report/Report.kt
package report

import java.io.*
import java.nio.charset.StandardCharsets
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

class Report/*report(ExternalEggSupplier externalEggSupplier, Customer customerArray,Product product, Egg egg) throws IOException {
		this.externalEggSupplier = externalEggSupplier;
		this.customerArray = customerArray;
		this.product = product;
		this.egg = egg;
	}*/
@Throws(IOException::class)
constructor() {
    private val date = DateTimeFormatter.ofPattern("EEEE MMMM dd, yyyy HH:mm:ss")
    private val now = LocalDateTime.now()
    private val writeTable = FileWriter("src/resources/report.txt", StandardCharsets.UTF_8)
    private val bw = BufferedWriter(writeTable)
    private val out = PrintWriter(bw)
    private val supplier = BufferedReader(FileReader("src/resources/Supplier Info.txt"))
    private val customer = BufferedReader(FileReader("src/resources/Customer Info.txt"))
    private val product = BufferedReader(FileReader("src/resources/Product.txt"))

    @Throws(IOException::class)
    fun generateReport() {
        out.println("Created by " + "Supervisor ")
        out.println("Compiled on: " + date.format(now) + "\n\n")
        val sup = supplier.readLines()
        val cust = customer.readLines()
        val prod = product.readLines()

        sup.forEach { out.println(it) }
        out.println("\n")
        cust.forEach { out.println(it) }
        out.println("\n")
        prod.forEach { out.print(it + "\n") }
        out.println("\n")

        /*out.println(sup)


        out.println(cust)
        out.println("\n")
        out.println(prod)*/

        out.flush()
        out.close()
        supplier.close()
        customer.close()
        product.close()
    }


}